app/src/main/java/br/com/nespressohelper/daitan/nespressohelper/BitmapHandler.java

```diff
@@ -19,37 +19,30 @@
     private Bitmap bitmap;
     private ByteArrayOutputStream stream;
     private byte[] bitmapData;
+    
 
-    public byte[] getImageBitmapData(Bitmap bitmap) {
-        if(bitmap == null) {
-            drawable = ContextCompat.getDrawable(App.getContext(), R.drawable.coffecapsule);
-            bitmap = ((BitmapDrawable) drawable).getBitmap();
-            stream = new ByteArrayOutputStream();
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
-            bitmapData = stream.toByteArray();
-            return bitmapData;
-        }else{
-            stream = new ByteArrayOutputStream();
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
-            bitmapData = stream.toByteArray();
-            return bitmapData;
-        }
-    }
+    public byte[] getImageBitmapData(int index, int length, TypedArray drawables, Bitmap bitmap) {
 
-    public byte[] getImageBitmapData(int index, int length, TypedArray drawables) {
-        if(index >= length) {
+        if(index >= length && bitmap == null){
             drawable = ContextCompat.getDrawable(App.getContext(), R.drawable.coffecapsule);
             bitmap = ((BitmapDrawable) drawable).getBitmap();
             stream = new ByteArrayOutputStream();
             bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
             bitmapData = stream.toByteArray();
             return bitmapData;
         }else{
-            drawable = ContextCompat.getDrawable(App.getContext(), drawables.getResourceId(index, -1));
-            bitmap = ((BitmapDrawable) drawable).getBitmap();
-            stream = new ByteArrayOutputStream();
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
-            return bitmapData = stream.toByteArray();
+            if(bitmap != null){
+                stream = new ByteArrayOutputStream();
+                bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+                bitmapData = stream.toByteArray();
+                return bitmapData;
+            }else{
+                drawable = ContextCompat.getDrawable(App.getContext(), drawables.getResourceId(index, -1));
+                bitmap = ((BitmapDrawable) drawable).getBitmap();
+                stream = new ByteArrayOutputStream();
+                bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+                return bitmapData = stream.toByteArray();
```

Comment:
You could create a method like:
```
public byte[] getByteArrayFromBitmap(Bitmap bitmap) {
    stream = new ByteArrayOutputStream();
    bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
    return bitmapData = stream.toByteArray();
}
```

So that this method would be:
```
public byte[] getImageBitmapData(int index, int length, TypedArray drawables, Bitmap bitmap) {
    if(bitmap == null){
       Drawable drawable; 
       if (index >= length) {
              drawable = ContextCompat.getDrawable(App.getContext(), R.drawable.coffecapsule);
        } else {
              drawable = ContextCompat.getDrawable(App.getContext(), drawables.getResourceId(index, -1));
        }
        bitmap = drawable.getBitmap():
    }
    return getByteArrayFromBitmap(bitmap);
}
```